### PR TITLE
Update test_api dependency, bump & fix lints, prepare release

### PR DESCRIPTION
**Pull request by kevmoo**
Closes https://github.com/dart-lang/matcher/pull/236

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
-  test_api: ">=0.5.0 <0.7.0"
+  test_api: ">=0.5.0 <0.8.0"
 
 dev_dependencies:
   fake_async: ^1.3.0
-  lints: ^2.0.0
+  lints: ^3.0.0
   test: ^1.23.0
 
 dependency_overrides:
```
Comment (jakemac53) [inline pubspec.yaml]: Can we drop the overrides?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,6 @@
-## 0.12.17-wip
+## 0.12.17
```
Comment (jakemac53) [inline CHANGELOG.md]: @natebosch thoughts on doing `0.12.16+1` to avoid a package:test release?

**Review by jakemac53 — APPROVED** [review]

File: pubspec.yaml (hunk 1 above)
Comment (kevmoo) [reply pubspec.yaml]: yes!

File: CHANGELOG.md (hunk 2 above)
Comment (natebosch) [reply CHANGELOG.md]: SGTM.

**Pull request merged**